File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task60_PERPLEXITY_llama-3-sonar-large-32k-chat.c
#include <stdio.h>
#include <string.h>

char* expandedForm(int num) {
    static char str[100];
    sprintf(str, "%d", num);
    char result[100] = "";
    int len = strlen(str);
    for (int i = 0; i < len; i++) {
        if (str[i] != '0') {
            strncat(result, &str[i], 1);
            for (int j = i + 1; j < len; j++) {
                strncat(result, "0", 1);
            }
            if (i < len - 1) {
                strncat(result, " + ", 3);
            }
        }
    }
    return str;
}

File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task60_PERPLEXITY_llama-3-sonar-large-32k-chat.java
public class Kata {
    public static String expandedForm(int num) {
        String str = String.valueOf(num);
        String result = "";
        for (int i = 0; i < str.length(); i++) {
            if (str.charAt(i) != '0') {
                result += str.charAt(i);
                for (int j = i + 1; j < str.length(); j++) {
                    result += "0";
                }
                if (i < str.length() - 1) {
                    result += " + ";
                }
            }
        }
        return result;
    }
}